src/openforms/payments/contrib/ogone/plugin.py

```diff
@@ -130,7 +132,12 @@ def handle_webhook(self, request):
 
         return payment
 
-    def apply_status(self, payment, ogone_status) -> None:
+    def apply_status(
+        self,
+        payment: SubmissionPayment,
+        ogone_status: str,
+        payment_id: str | None = None,
```

Comment:
in general we try to avoid type unions if you can do the same thing with a single type:

```suggestion
        payment_id: str = "",
```
would be nicer

Do I understand it correctly that `payment_id` is not guaranteed to be non-empty-ish because the user could cancel the payment flow, leading to no `PAYID` value being provided?

Reply:
`apply_status` is also used by `handle_webhook` and I'm not sure at what point in the flow that is called, though in the test I do see that PAYID is passed as part of the request body there, so maybe we can assume that it is / should be filled?

Reply:
yeah the webhook fires to ensure that we get an update, even if the end-user closes the browser tab (for example). I think it should get the same-ish parameters.